presto-main/src/main/java/com/facebook/presto/matching/MatchingEngine.java

```diff
@@ -43,7 +45,11 @@ private MatchingEngine(ListMultimap<Class, T> matchablesByClass)
 
     private static Iterator<Class> ancestors(Class clazz)
     {
-        return TreeTraverser.using((Class n) -> asList(Optional.ofNullable(n.getSuperclass())))
+        return TreeTraverser.using(
```

Comment:
the `supertypes` method I mentioned includes the interfaces. I'd use it here and drop the previous commit (sorry! :))